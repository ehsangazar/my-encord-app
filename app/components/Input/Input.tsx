import { ErrorMessage, Field, FieldProps } from "formik";

interface InputProps {
  name: string;
  type: string;
  placeholder: string;
}

const Input = ({ name, type, placeholder }: InputProps) => {
  return (
    <Field name={name}>
      {({ field }: FieldProps) => (
        <div className="flex flex-col w-full mb-4">
          <input
            className="border border-gray-300 rounded p-2 w-full"
            {...field}
            type={type}
            placeholder={placeholder}
          />
          <ErrorMessage name={name}>
            {(msg) => <div className="text-xs text-red-500">{msg}</div>}
          </ErrorMessage>
        </div>
      )}
    </Field>
  );
};

export default Input;
