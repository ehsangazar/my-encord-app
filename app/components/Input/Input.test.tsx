import renderer from "react-test-renderer";

import Input from "./Input";
import { Formik } from "formik";

it("Input component should render correctly", () => {
  const tree = renderer
    .create(
      <Formik initialValues={{ name: "" }} onSubmit={() => {}}>
        <Input name="name" type="text" placeholder="Name" />
      </Formik>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
