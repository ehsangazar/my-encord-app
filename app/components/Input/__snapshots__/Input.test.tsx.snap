// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render correctly 1`] = `
<div
  className="flex flex-col w-full mb-4"
>
  <input
    className="border border-gray-300 rounded p-2 w-full"
    name="name"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
</div>
`;
